GeometricSeqPhenotype.java

```diff
@@ -0,0 +1,308 @@
+     *
+     * @return position of this GeometricSeqPhenotype along the y-axis
+     */
+    public double getTraitB() {
+        return traitB;
+    }
+
+    public String getSequence() {
+        return String.valueOf(this.nucleotideSequence);
+    }
+
+    /**
+     * Return the (Euclidean) distance between this GeometricSeqPhenotype and p in Euclidean space
+     *
+     * @param p  GeometricSeqPhenotype to calculate the distance between
+     * @return distance between this GeometricSeqPhenotype and p
+     */
+    public double distance(Phenotype p) {
+        GeometricSeqPhenotype p2d = (GeometricSeqPhenotype) p;
+        double distA = (getTraitA() - p2d.getTraitA());
+        double distB = (getTraitB() - p2d.getTraitB());
+        double dist = (distA * distA) + (distB * distB);
+        dist = Math.sqrt(dist);
+        checkRep();
+        return dist;
+    }
+
+    /**
+     * Returns a mutated copy of this GeometricSeqPhenotype (point substitution), original GeometricSeqPhenotype is unharmed
+     *
+     * @return a mutated copy of this GeometricSeqPhenotype
+     */
+    public Phenotype mutate() {
+        // Implementation:
+        // Mutates a nucleotide in the sequence at a random index
+        // If mutation creates a stop codon, then mutate at another index
+
+        // this.nucleotideSequence is represented using a char[] instead of
+        //  - String since Strings are immutable. New Strings have to be created to mutate a sequence,
+        //    which is messy and uses up extra heap space. Plus, Strings include an extra byte '\n'.
+        //  - StringBuffer because it's just a Wrapper around a char[], and has functionality that we don't need such as resizing.
+
+        String wildTypeAminoAcid = "", mutantAminoAcid = "";
+        int mutationIndexSiteSequence = -1;
+        char mutatedNucleotide = ' ', originalNucleotide = ' ';
+
+        // mutate at least once
+        // keep mutating if mutation generates a stop codon
+        while (mutantAminoAcid.equals("") || mutantAminoAcid.equals("STOP")) {
+            // choose random index to mutate in this.nucleotideSequence
+            mutationIndexSiteSequence = random(this.nucleotideSequence.length);
+
+            originalNucleotide = this.nucleotideSequence[mutationIndexSiteSequence];
+
+            char originalNucleotideToMutate = this.nucleotideSequence[mutationIndexSiteSequence];
+
+            // get mutant nucleotide (transition transversion ratio)
+            mutatedNucleotide = Biology.MutationType.MUTATION.getNucleotide(originalNucleotideToMutate);
+
+            String[] wildTypeMutantAminoAcids = mutateHelper(mutationIndexSiteSequence, mutatedNucleotide);
+
+            wildTypeAminoAcid = wildTypeMutantAminoAcids[0];
+            mutantAminoAcid = wildTypeMutantAminoAcids[1];
+        }
+
+        int mutationIndexSite = mutationIndexSiteSequence / 3; // site # where mutation is occurring {0, . . ., total number of sites - 1}
```

Comment:
To make the code clearer, would it be easy to rename these variables according to whether they refer to the nucleotide sequence or amino-acid sequence?